src/wp-includes/functions.php

```diff
@@ -7695,3 +7695,120 @@ function is_php_version_compatible( $required ) {
 function wp_fuzzy_number_match( $expected, $actual, $precision = 1 ) {
 	return abs( (float) $expected - (float) $actual ) <= $precision;
 }
+
+/**
+ * Formats attributes string from associative array $attributes to be placed inside a `<script>` tag.
+ *
+ * Automatically injects type attribute if needed.
+ * Used by wp_get_script_tag() and wp_get_inline_script_tag().
+ *
+ * @since 5.6.0
+ *
+ * @param array $attributes Optional. Key-value pairs representing `<script>` tag attributes.
+ * @return string String made of sanitized `<script>` tag attributes.
+ */
+function wp_sanitize_script_attributes( $attributes = array() ) {
+	if ( ! array_key_exists( 'type', $attributes ) && ! is_admin() && ! current_theme_supports( 'html5', 'script' ) ) {
+		$attributes['type'] = 'text/javascript';
+	}
+	/**
+	 * Filters script attributes.
+	 *
+	 * $attributes can be modified to add attributes into the script tag and to change or delete existing ones.
+	 *
+	 * @since 5.6.0
+	 *
+	 * @param array $attributes Key-value pairs representing `<script>` tag attributes.
+	 */
+	$attributes = apply_filters( 'wp_script_attributes', $attributes );
+
+	$attributes_string = '';
+	// Only the attribute name is added to $attributes_string for entries with a boolean value, and that are true.
+	foreach ( $attributes as $attribute_name => $attribute_value ) {
+		if ( is_bool( $attribute_value ) ) {
+			if ( $attribute_value && '' !== $attribute_name ) {
```

Comment:
Is the `'' !== $attribute_name` really needed here? I'd say either we should add the check regardless of whether or not the value is a boolean, or we should omit it (expecting correct usage of the function). If the attribute name is empty, it would always be invalid usage of the function.

Reply:
I agree, l will remove it.